File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    float change;

    // ask the user for input
    do
    {
        change = get_float("Enter the amount of change owed: $");
    }
    while (change < 0);

    // convert the amount to cents (float to int)
    int cents = 100 * change;
    // but make sure that fractional sets are rounded, not truncated
    if ((100 * change - cents) >= 0.005)
    {
        ++cents;
    }

    // calculate the quarters
    int coins = cents / 25;
    cents -= 25 * coins;

    // calculate the dimes
    coins += cents / 10;
    cents -= 10 * (cents / 10);

    // calculate the nickels
    coins += cents / 5;

    // calculate the pennies
    coins += cents % 5;

    // produce the output
    printf("%i\n", coins);
}

File: pset6/cash/cash.py
from cs50 import get_float

# prompt for user input
while True:
    change = 100 * get_float("Change owed: ")
    if not change < 0:
        break

# calculate quarters...
coins = change // 25
change = change % 25

# dimes...
coins += change // 10
change = change % 10

# nickles...
coins += change // 5
change = change % 5

# ... and pennies
coins += change // 1

# output result
print(coins)